airflow/ui/src/queries/useClearTaskInstances.ts

```diff
@@ -60,13 +60,21 @@ export const useClearTaskInstances = ({
       onSuccessDryRun(data);
     } else {
       const taskInstanceKeys = (variables.requestBody.task_ids ?? [])
+        .filter(
+          (taskId): taskId is string | [string, number] =>
+            typeof taskId === "string" || Array.isArray(taskId),
+        )
         .map((taskId) => {
+          // Determine the actual task_id
+          const actualTaskId = Array.isArray(taskId) ? taskId[0] : taskId;
```

Comment:
As we implemented the fallback approach, we need to deduplicate here. because `taskId[0]` could possibly be the same task_id.

And we don't want to pass duplicated key to invalidate cache.

Reply:
Also we need to adapt the front-end code to support a single map-index clearing. That can be a follow up PR though.